### PR TITLE
Social media

### PR DESCRIPTION
**Pull request by bwormser**
Added Social media forum page.

Notes:
Gonna make Marco's metric code a component to use in these once that's pushed

### CODE REVIEW

**Comment by roccopolimen** [comment]
I'll wait til marco's code gets used and merged to look at this

File: client/src/api/routes/users.ts [hunk 1]
```diff
@@ -89,6 +89,20 @@ export const useDeleteUser = (): UseQueryResult<boolean> => {
     });
 };
 
+/**
+ * @description GET /users/:userId
+ * @param {string} userId 
+ * @returns {UseQueryResult<UserObject>} the requested user
+ * @throws if fails
+ */
+ export const useGetUser = (userId: string): UseQueryResult<UserObject> => {
```
Comment (roccopolimen) [inline client/src/api/routes/users.ts]: we need to get rid of this

File: client/src/pages/Forum/index.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import Loading from 'components/Loading';
+
+export default function Forum() {
+    let startPosts: ForumPostObject[] = [];
+
+    const [searchParams, setSearchParams] = useSearchParams();
+    const [posts, setPosts] = useState(startPosts);
+    const [loading, setLoading] = useState(true);
+    const [refresh, setRefresh] = useState(true);
+    const [numPosts, setNumPosts] = useState(3);
```
Comment (roccopolimen) [inline client/src/pages/Forum/index.tsx]: hard coded 3?

File: server/routes/users.ts [hunk 3]
```diff
@@ -109,4 +110,19 @@ router.delete('/', async (req, res) => {
     }
 });
 
+//GET /:id
+router.get('/:userId', async (req, res) => {
```
Comment (roccopolimen) [inline server/routes/users.ts]: remove

**Review by roccopolimen — COMMENTED** [review]
Other than pushing the env files (which I removed already with GitHub), need to remove the hook and route for getting the user information so it is not publicly available.

File: client/src/api/routes/users.ts (hunk 1 above)
Comment (bwormser) [reply client/src/api/routes/users.ts]: Yep I deleted these last night but forgot to push but they should be gone now

File: server/routes/users.ts (hunk 3 above)
Comment (bwormser) [reply server/routes/users.ts]: Same thing as above this should be gone now

File: client/src/pages/Forum/index.tsx (hunk 2 above)
Comment (bwormser) [reply client/src/pages/Forum/index.tsx]: So yeah in order to give it the number typing while still using the api call I gave it a default of 3, I changed this to be a '0' string now which defaults to getting all the posts if the num_posts parameter isn't specified. If we want something else i can change it

**Comment by GraceMattern** [comment]
LGTM

**Pull request merged**